src/helperFunctions/hash.py

```diff
@@ -51,27 +62,25 @@ def get_tlsh_comparison(first, second):
     return tlsh.diff(first, second)  # pylint: disable=c-extension-no-member
 
 
-def get_imphash(file_object):
+def get_imphash(file_object: FileObject) -> str | None:
     '''
-    Generates and returns the md5 hash of the imported functions of an ELF file
-    represented by `file_object`.
-    The imports are sorted before the hex is generated so the order of imports
-    does not matter.
+    Generates and returns the md5 hash of the (sorted) imported functions of an ELF file represented by `file_object`.
+    Returns `None` if there are no imports or if an exception occurs.
 
     :param file_object: The FileObject of which the imphash shall be computed
     '''
     if _is_elf_file(file_object):
         try:
             with _suppress_stdout():
-                # pylint: disable=c-extension-no-member
-                functions = normalize_lief_items(lief.parse(file_object.file_path).imported_functions)
-            return md5(','.join(sorted(functions)).encode()).hexdigest()
-        except Exception:
+                functions = [f.name for f in lief.ELF.parse(file_object.file_path).imported_functions]
```

Comment:
How exactly does this fix the error that we got before?

Reply:
The problematic binaries that caused the segfaults were [android OAT binaries](https://www.droidwiki.org/wiki/OAT) and lief has a separate parser for these files. When calling `lief.parse()` the parser is chosen automatically by lief. The segfault was caused by lief's OAT parser. Using `lief.ELF.parse()` forces the use of the ELF parser (which does not crash).